content/blogs/YYYY-MM-DD-CI-CD-kestra-comparison.md

```diff
@@ -0,0 +1,186 @@
+
+In fact, if your workflow goes beyond just automating builds and deployments  — let’s say you need to automate data processing tasks across cloud services — here's where Kestra’s orchestration capabilities really shine. Kestra, in fact, can handle more complex workflows with dependencies between services, and it also provides mechanisms for error handling and managing events - even [in real time](./2024-06-27-realtime-triggers.md).
+
+So, if your use case involves coordinating tasks beyond GitHub, Kestra becomes a better choice than GitHub Actions.
+
+### Kestra Vs. GitLab CI/CD
+Like GitHub Actions, [GitLab CI/CD](https://docs.gitlab.com/ee/ci/index.html) is a great tool for CI/CD. It's integrated directly with GitLab and works well for automating pipelines from code commits to production. It also scales nicely, especially if you’re already in the GitLab ecosystem.
+
+![GitLab CI/CD by Federico Trotta](/blogs/YYYY-MM-DD-CI-CD-kestra-comparison/gitlab_cicd.png)
+
+Although GitLab CI/CD is more suited for teams needing advanced pipeline capabilities and integrated DevOps tooling than GitHub Actions, it basically has the same weakness as GitHub Actions when the game becomes harder.
+
+In fact, when you need something that can handle orchestration across multiple environments, or when workflows involve more than just code deployments, Kestra’s orchestration integrations take over.
+
+For example: do you need to coordinate a workflow that involves running processes across different cloud providers or managing task dependencies in a data pipeline? That’s where Kestra outperforms GitLab CI/CD which is great for automating code deployments, but Kestra is designed to manage more intricate workflows that involve different systems and cloud environments.
+
+
+### Kestra Vs. Azure DevOps
+Azure DevOps is another heavy hitter for CI/CD, particularly if you’re in the Microsoft ecosystem. It’s a fully integrated toolchain that works well for automating pipelines, managing repos, and monitoring applications. 
+
+![Azure DevOps workflow by Federico Trotta](/blogs/YYYY-MM-DD-CI-CD-kestra-comparison/azure_devops.png)
+
+But again, its core strength lies in CI/CD, not in orchestrating workflows across multiple systems or environments.
+
+For example, let’s say you’re managing a workflow that includes tasks across multiple cloud providers, and Azure is just one part of the puzzle. Maybe you’re also pulling in data from AWS or triggering workflows based on events in GCP (Google Cloud Platform). In this case, Kestra allows you to orchestrate these tasks in a single workflow, regardless of which system is involved. 
+
+So, while Azure DevOps is great for Microsoft-centric pipelines, Kestra offers the flexibility to handle distributed, cross-cloud workflows with ease.
+
+
+### Kestra vs. Jenkins

```

Comment:
I'd say the main thing Kestra does well here is the simple interface with good autocomplete and error handling to make building your system quick. I use to work with Jenkins and it was a very slow and tedious process to get setup in comparison.